apps/tangle-cloud/src/pages/instances/Instances/UpdateBlueprintModel/ApproveConfirmationModal.tsx

```diff
@@ -0,0 +1,263 @@
+            };
+          }
+        }
+      });
+
+      values.requestId = selectedRequest?.requestId as number;
+
+      return {
+        values,
+        errors: Object.keys(errors).length > 0 ? errors : {},
+      };
+    },
+  });
+
+  const securityCommitment = watch('securityCommitment');
+
+  const onCancel = () => {
+    reset({
+      requestId: selectedRequest?.requestId,
+      securityCommitment: [
+        {
+          assetId: '',
+          exposurePercent: '',
+        },
+      ],
+    });
+    onClose();
+  };
+
+  const onSubmit = async (data: ApprovalConfirmationFormFields) => {
+    const isSuccess = await onConfirm(data);
+    if (!isSuccess) return;
+
+    reset({
+      requestId: selectedRequest?.requestId,
+      securityCommitment: [
+        {
+          assetId: '',
+          exposurePercent: '',
+        },
+      ],
+    });
+    onClose();
+  };
+
+  return (
+    <ModalContent
+      size="lg"
+      onInteractOutside={(event) => event.preventDefault()}
+      title={`Service Request #${selectedRequest?.requestId}`}
+      description="Are you sure you want to approve this blueprint?"
+    >
+      <ModalHeader>Service Request #{selectedRequest?.requestId}</ModalHeader>
+
+      <ModalBody>
+        <BlueprintItem
+          imgUrl={selectedRequest?.blueprintData?.metadata.logo ?? ''}
+          renderImage={(imageUrl) => {
+            return (
+              <img
+                src={imageUrl}
+                alt={selectedRequest?.blueprintData?.metadata.name ?? ''}
+                className="flex-shrink-0 bg-center rounded-full"
+              />
+            );
+          }}
+          id={selectedRequest?.blueprintData?.metadata.name ?? ''}
+          name={selectedRequest?.blueprintData?.metadata.name ?? ''}
+          restakersCount={selectedRequest?.blueprintData?.restakersCount ?? 0}
+          operatorsCount={selectedRequest?.blueprintData?.operatorsCount ?? 0}
+          tvl={selectedRequest?.blueprintData?.tvl?.toString() ?? '0'}
+          isBoosted={false}
+          category={selectedRequest?.blueprintData?.metadata.category ?? ''}
+          description={
+            selectedRequest?.blueprintData?.metadata.description ?? ''
+          }
+          author={selectedRequest?.blueprintData?.metadata.author ?? ''}
+        />
+
+        <form
+          ref={formRef}
+          onSubmit={handleSubmit(onSubmit)}
+          className="mt-4 space-y-4"
+        >
+          {Children.toArray(
+            securityCommitment.map((_, index) => (
+              <AssetCommitmentFormItem
+                index={index}
+                assetId={watch(`securityCommitment.${index}.assetId`)}
+                assetMetadata={assetsMetadata}
+                assetOptions={selectedRequest?.securityRequirements}
+                onChangeAssetId={(value) => {
+                  setValue(
+                    `securityCommitment.${index}.assetId`,
+                    value as RestakeAssetId,
```

Comment:
Use `assertRestakeAssetId` here to help catch logic bugs at runtime:
```suggestion
                    assertRestakeAssetId(value),
```